native/src/modules/layout/__tests__/HeaderContainer.spec.js

```diff
@@ -102,4 +105,118 @@ describe('HeaderContainer', () => {
       expect.objectContaining({ shareUrl: url('abc') })
     )
   })
+
+  it('shareUrl should be set correctly for events route', () => {
+    jest.doMock('../components/Header', () => Header)
+    const HeaderContainer = require('../containers/HeaderContainer').default
+
+    const eventsRouteMapping = {
+      routeKeyEvent1: {
+        status: 'ready',
+        path: null,
+        language: language._code,
+        city: city.name,
+        models: [],
+        allAvailableLanguages: new Map()
+      }
+    }
+
+    const state: StateType = prepareState(eventsRouteMapping, {}, {})
+    const store = mockStore(state)
+
+    const ownProps = {
+      scene: {
+        route: {
+          name: EVENTS_ROUTE,
+          key: 'routeKeyEvent1'
+        }
+      }
+    }
+
+    const result = TestRenderer.create(
+      <Provider store={store}>
+        {/* $FlowFixMe not all props passed */}
+        <HeaderContainer {...ownProps}/>
+      </Provider>
+    )
+
+    const header = result.root.findByType(Header)
+
+    expect(header.props).toEqual(
+      expect.objectContaining({ shareUrl: cityContentUrl({ cityCode: city.code, languageCode: language._code, route: EVENTS_ROUTE, path: null }) })
+    )
+  })
+
+  it('shareUrl should be set correctly for local news route', () => {
+    jest.doMock('../components/Header', () => Header)
+    const HeaderContainer = require('../containers/HeaderContainer').default
+
+    const newsRouteMapping = {
+      routeKeyNews1: {
+        status: 'ready',
+        models: [],
+        hasMoreNews: false,
+        page: 1,
+        newsId: null,
+        language: language._code,
+        city: city.name,
+        type: LOCAL_NEWS_TYPE,
+        allAvailableLanguages: new Map()
+      }
+    }
+
+    const state: StateType = prepareState({}, {}, newsRouteMapping)
+    const store = mockStore(state)
+
+    const ownProps = {
+      scene: {
+        route: {
+          name: NEWS_ROUTE,
+          key: 'routeKeyNews1'
+        }
+      }
+    }
+
+    const result = TestRenderer.create(
+      <Provider store={store}>
+        {/* $FlowFixMe not all props passed */}
+        <HeaderContainer {...ownProps}/>
+      </Provider>
+    )
+
+    const header = result.root.findByType(Header)
+
+    expect(header.props).toEqual(
+      expect.objectContaining({ shareUrl: cityContentUrl({ cityCode: city.code, languageCode: language._code, route: NEWS_ROUTE, path: LOCAL_NEWS_TYPE }) })
+    )
+  })
+
+  it('shareUrl should be set correctly for offers route', () => {
```

Comment:
Tests for pois and disclaimer are missing, could you add some? I think you could also generalize this kind of test by creating helper functions to render and assert. I think it is also not necessary to create the state separately each time, you can just set all mappings.

Reply:
added tests for disclaimer and pois and also added a render and assert function and you are completly right, now it is a lot simpler. :)